supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategy.scala

```diff
@@ -87,10 +91,61 @@ final case class HashPathModTimeStrategy(checkSiblingMd5: Boolean) extends Confi
   override val description = "hash file path and last modified time"
 }
 
-final case class HashFileStrategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+final case class HashFileMd5Strategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
   override protected def hash(file: Path): Try[String] = {
     tryWithResource(() => file.newInputStream) { DigestUtils.md5Hex }
   }
 
-  override val description = "hash file content"
+  override val description = "hash file content with md5"
 }
+
+final case class HashFileXxH64Strategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+  override protected def hash(file: Path): Try[String] = {
+    tryWithResource(() => file.newInputStream) {HashFileXxH64StrategyMethods.xxh64sum(_)}
+  }
+  override val description = "hash file content with xxh64"
+}
+
+final case class FingerprintStrategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+  override protected def hash(file: Path): Try[String] = {
+    Try {
+      file.lastModifiedTime.toEpochMilli.toHexString +
+      file.size.toHexString +
+      // Only check first 10 MB (10485760 bytes) for performance reasons
+      HashFileXxH64StrategyMethods.xxh64sum(file.newInputStream, maxSize = 10485760L)
```

Comment:
Is it worth having a secondary digest to produce a fixed-length string. Right now this result is variable length which seems potentially a little strange

Reply:
:+1: Yes. I created a xxh64sumString method to calculate the xxh64 hash from a string. Should be implemented now.